modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/TracingLevelExportInfo.java

```diff
@@ -9,64 +9,98 @@
 import org.mule.runtime.api.component.Component;
 import org.mule.runtime.tracer.api.span.info.InitialExportInfo;
 import org.mule.runtime.tracer.customization.api.InitialExportInfoProvider;
+import org.mule.runtime.tracer.customization.impl.info.SpanInitialInfoUtils;
+import org.mule.runtime.tracer.customization.impl.provider.DebugInitialExportInfoProvider;
+import org.mule.runtime.tracer.customization.impl.provider.MonitoringInitialExportInfoProvider;
+import org.mule.runtime.tracer.customization.impl.provider.OverviewInitialExportInfoProvider;
+import org.mule.runtime.tracing.level.api.config.TracingLevel;
+import org.mule.runtime.tracing.level.api.config.TracingLevelConfiguration;
 
 import java.util.Set;
 
-public class TracingLevelExportInfo {
+import static java.util.Collections.emptySet;
 
-  private InitialExportInfoProvider initialExportInfoProvider;
-  private Object spanIdentifier;
-  private boolean isOverride;
+public class TracingLevelExportInfo implements InitialExportInfo {
+
+  private final String spanName;
+  private TracingLevel tracingLevel;
   private InitialExportInfo initialExportInfo;
+  private Set<String> propagatedNoExportUntil = emptySet();
 
-  public TracingLevelExportInfo(InitialExportInfoProvider initialExportInfoProvider, boolean isOverride) {
-    this.initialExportInfoProvider = initialExportInfoProvider;
-    this.isOverride = isOverride;
+  public TracingLevel getTracingLevel() {
+    return tracingLevel;
   }
 
-  public void setSpanIdentifier(Component spanIdentifier) {
-    this.spanIdentifier = spanIdentifier;
+  private TracingLevelExportInfo(TracingLevel tracingLevel, String spanName) {
+    this.tracingLevel = tracingLevel;
+    this.spanName = spanName;
   }
 
-  public void setSpanIdentifier(String spanIdentifier) {
-    this.spanIdentifier = spanIdentifier;
+  public static TracingLevelExportInfo createTracingLevelExportInfo(Component component, String spanName,
+                                                                    TracingLevelConfiguration tracingLevelConfiguration) {
+    return new TracingLevelExportInfo(tracingLevelConfiguration
+        .getTracingLevel(SpanInitialInfoUtils.getLocationAsString(component.getLocation())), spanName);
   }
 
-  public boolean isOverride() {
-    return this.isOverride;
+  private static InitialExportInfoProvider resolveInitialExportInfoProvider(TracingLevel tracingLevel) {
+    switch (tracingLevel.getTracingLevelId()) {
```

Comment:
"JUST AN IDEA": Maybe the tracingLevel can know which export provider should be used based on its tracingLevelId without exposing it. Then the other uses of getTracingLevelId() will be only for testing purposes and probably can be removed.